tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -464,6 +471,8 @@ CGMSHLSLRuntime::SetSemantic(const NamedDecl *decl,
     if (it->getKind() == hlsl::UnusualAnnotation::UA_SemanticDecl) {
       const hlsl::SemanticDecl *sd = cast<hlsl::SemanticDecl>(it);
       paramInfo.SetSemanticString(sd->SemanticName);
+      if (m_PreciseOutputSet.count(sd->SemanticName))
```

Comment:
I believe we will want to be case-insensitive, since semantic names are.  And, how is semantic index handled?